hmda-analytics/src/main/scala/hmda/analytics/HmdaAnalyticsApp.scala

```diff
@@ -66,118 +66,105 @@ object HmdaAnalyticsApp extends App with TransmittalSheetComponent with LarCompo
    * In beta namespace this environment variable has currentSchema=hmda_beta_user appended to it to change the schema
    * to BETA
    */
-  val tsTableName2018  = config.getString("hmda.analytics.2018.tsTableName")
-  val larTableName2018 = config.getString("hmda.analytics.2018.larTableName")
+
+  case class YearlyTransmittalSheetRepositoryWrapper(year: String){
+    val transmittalSheet = new TransmittalSheetRepository(dbConfig, getTableNameByYear)
+
+    def getTransmittalSheet = transmittalSheet
+
+    def getId(row: TransmittalSheetTable): Unit = transmittalSheet.getId(row)
+    def createSchema(): Unit = transmittalSheet.createSchema()
+    def dropSchema(): Unit = transmittalSheet.dropSchema()
+
+    def insert(ts: TransmittalSheetEntity): Future[Int] = transmittalSheet.insert(ts)
+
+    def findByLei(lei: String): Future[Seq[TransmittalSheetEntity]] = transmittalSheet.findByLei(lei)
+
+    def deleteByLei(lei: String): Future[Int] = transmittalSheet.deleteByLei(lei)
+
+    def deleteByLeiAndQuarter(lei: String): Future[Int] = transmittalSheet.deleteByLeiAndQuarter(lei)
+
+    def updateByLei(ts: TransmittalSheetEntity): Future[Int] = transmittalSheet.updateByLei(ts)
+
+    def count(): Future[Int] = transmittalSheet.count()
+
+    private def getTableNameByYear = {
+      val configString = s"hmda.analytics.$year.tsTableName"
+      config.getString(configString)
+    }
+
+  }
+
+  case class QuarterlyTransmittalSheetRepositoryWrapper(year: String, quarter: String){
+    val transmittalSheet = new TransmittalSheetRepository(dbConfig, getTableNameByYear)
+
+    def getTransmittalSheet = transmittalSheet
+
+
+    def getId(row: TransmittalSheetTable): Unit = transmittalSheet.getId(row)
+    def createSchema(): Unit = transmittalSheet.createSchema()
+    def dropSchema(): Unit = transmittalSheet.dropSchema()
+
+    def insert(ts: TransmittalSheetEntity): Future[Int] = transmittalSheet.insert(ts)
+
+    def findByLei(lei: String): Future[Seq[TransmittalSheetEntity]] = transmittalSheet.findByLei(lei)
+
+    def deleteByLei(lei: String): Future[Int] = transmittalSheet.deleteByLei(lei)
+
+    def deleteByLeiAndQuarter(lei: String): Future[Int] = transmittalSheet.deleteByLeiAndQuarter(lei)
```

Comment:
So how do transmittalSheet.deleteByLeiAndQuarter (and larRepo.deletebyLeiAndQuarter below) actually work if we don't call them with a quarter?

Reply:
Both were probably needed at some point, the second one just further filters by quarterly records. This function is no longer needed especially after the refactor, will delete.